ci/Jenkinsfiles/javadoc.groovy

```diff
@@ -0,0 +1,278 @@
+        }
+      }
+    }
+
+    stage('Deploy Nuxeo ECM Javadoc') {
+      when {
+        branch 'PR-*'
+        expression {
+          return pullRequest.labels.contains('preview-javadoc')
+        }
+        not {
+          environment name: 'DRY_RUN', value: 'true'
+        }
+      }
+      steps {
+        setGitHubBuildStatus('javadoc/preview', 'Deploy Javadoc environment', 'PENDING')
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Build Nuxeo ECM Javadoc Docker Image ${VERSION}
+          ----------------------------------------
+          Image tag: ${VERSION}"""
+          sh "mv target/site/apidocs ci/docker/javadoc/apidocs"
+          dir('ci/docker/javadoc') {
+            sh "skaffold build -f skaffold.yaml"
+          }
+
+          echo """
+          ----------------------------------------
+          Deploy Nuxeo ECM Javadoc Environment
+          ----------------------------------------
+          Image tag: ${VERSION}
+          Namespace: ${PREVIEW_NAMESPACE}"""
+          dir('ci/helm/javadoc') {
+            script {
+              // first substitute environment variables in chart values
+              sh """
+                mv values.yaml values.yaml.tosubst
+                envsubst < values.yaml.tosubst > values.yaml
+              """
+              // second create target namespace (if doesn't exist) and copy secrets to target namespace
+              boolean nsExists = sh(returnStatus: true, script: "kubectl get namespace ${PREVIEW_NAMESPACE}") == 0
+              if (!nsExists) {
+                sh "kubectl create namespace ${PREVIEW_NAMESPACE}"
+              }
+              sh "kubectl --namespace platform get secret kubernetes-docker-cfg -ojsonpath='{.data.\\.dockerconfigjson}' | base64 --decode > /tmp/config.json"
+              sh """kubectl create secret generic kubernetes-docker-cfg \
+                  --namespace=${PREVIEW_NAMESPACE} \
+                  --from-file=.dockerconfigjson=/tmp/config.json \
+                  --type=kubernetes.io/dockerconfigjson --dry-run -o yaml | kubectl apply -f -"""
+              // third build and deploy the chart
+              sh """
+                jx step helm build --verbose
+                jx step helm install --namespace ${PREVIEW_NAMESPACE} --name javadoc --verbose .
+              """
+              host = sh(returnStdout: true, script: "kubectl get ingress --namespace=${PREVIEW_NAMESPACE} javadoc -ojsonpath='{.items[*].spec.rules[*].host}'")
+              echo """
+              ----------------------------------------
+              Javadoc Environment available at: https://${host}
+              ----------------------------------------"""
+              // comment the PR if it is the first time
+              if (isPullRequest() && !nsExists) {
+                pullRequest.comment("Preview Javadoc environment available [here](https://${url}).")
```

Comment:
no need of JX :ok_hand:

Reply:
Indeed 🙂, you make me think that I don't handle the namespace cleanup. I'll have a look to do it without the jx command.

Reply:
In the end we're gonna need the jx preview command, but I will keep this block and disable comment on jx command.